stbt-camera.d/gst/stbtcontraststretch.c

```diff
@@ -146,30 +146,42 @@ stbt_contrast_stretch_finalize (GObject * object)
 static GstSample*
 stbt_contrast_stretch_load_png (const gchar *filename, const GstCaps *caps)
 {
-  GstElement *src, *sink;
-  GstSample *out;
-  GstElement *pipeline = gst_parse_launch (
-    "filesrc name=src ! pngdec ! videoconvert ! appsink name=sink", NULL);
+  GstElement *src = NULL, *sink = NULL;
+  GstSample *out = NULL;
+  GstElement *pipeline = NULL;
 
-  g_return_val_if_fail (pipeline, NULL);
+  pipeline = gst_parse_launch (
+    "filesrc name=src ! pngdec ! videoconvert ! appsink name=sink", NULL);
+  if (!pipeline)
+    goto exit;
 
   src = gst_bin_get_by_name(GST_BIN(pipeline), "src");
   sink = gst_bin_get_by_name(GST_BIN(pipeline), "sink");
-
-  g_return_val_if_fail (src && sink, NULL);
+  if ((src == NULL) || (sink == NULL))
+    goto exit;
   g_object_set(src, "location", filename, NULL);
   g_object_set(sink, "caps", caps, NULL);
 
   gst_element_set_state(GST_ELEMENT(pipeline), GST_STATE_PLAYING);
 
-  /* FIXME: deal with error conditions like missing file, etc. */
   out = gst_app_sink_pull_preroll(GST_APP_SINK (sink));
+  if (out == NULL)
+    goto exit;
+  if (!GST_IS_SAMPLE(out)) {
+    gst_sample_unref(out);
+    out = NULL;
+    goto exit;
+  }
```

Comment:
No, I mean that it needs less checking.   You can remove:

    + if (!GST_IS_SAMPLE(out)) {
    +    gst_sample_unref(out);
    +    out = NULL;
    +    goto exit;
    +  }

as it should never happen.